addons/controllers/webhooks/webhooks.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package webhooks provides functions to manage webhook TLS certificates
+package webhooks
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks"
+)
+
+type WebhookTLS struct {
+	Ctx           context.Context
+	K8sConfig     *rest.Config
+	CertPath      string
+	KeyPath       string
+	Name          string
+	ServiceName   string
+	LabelSelector string
+	Logger        logr.Logger
+	TLSSecret     *corev1.Secret
+	Namespace     string
+	InvalidWithIn time.Duration
+}
+
+func (w *WebhookTLS) Manage() error {
```

Comment:
Do we expect this to be used by callers directly or is an internal API?

Reply:
Callers will use it directly. For example, ManageTLSCertificates, is a caller and will only have access to it if we make expose it. 
(I think that is what you mean).

Reply:
Renamed Mange -> UpdateAndCreate to remove ambiguity. It really does not manage. It updates or creates. 
The caller to this function will be the manager.